ComponentKit/TransactionalDataSources/Common/CKTransactionalComponentDataSourceAppliedChanges.mm

```diff
@@ -0,0 +1,67 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import "CKTransactionalComponentDataSourceAppliedChanges.h"
+#import "CKTransactionalComponentDataSourceAppliedChanges+Internal.h"
+
+#import "ComponentUtilities.h"
+#import "CKInternalHelpers.h"
+#import "CKMacros.h"
+
+@implementation CKTransactionalComponentDataSourceAppliedChanges
+
+- (instancetype)initWithUpdatedIndexPaths:(NSSet *)updatedIndexPaths
+                        removedIndexPaths:(NSSet *)removedIndexPaths
+                          removedSections:(NSIndexSet *)removedSections
+                          movedIndexPaths:(NSDictionary *)movedIndexPaths
+                         insertedSections:(NSIndexSet *)insertedSections
+                       insertedIndexPaths:(NSSet *)insertedIndexPaths
+                                userInfos:(NSArray *)userInfos
```

Comment:
I'd actually like to argue for userInfos.

- They allow you to pipe data through and have the data source hang onto it, freeing you from needing to maintain a lookaside table.
- They are idiomatic to Objective-C.
- It matches our internal "connection store" class.

We can debate this in person. :)